packages/apputils-extension/src/index.ts

```diff
@@ -60,8 +60,8 @@ function apiError(id: string, xhr: XMLHttpRequest): Error {
 /**
  * Create a data connector to access plugin settings.
  */
-function newConnector(manager: IServiceManager): IDataConnector<ISettingRegistry.IPlugin, JSONObject> {
-  return {
+function newConnector(manager: IServiceManager): Promise<IDataConnector<ISettingRegistry.IPlugin, JSONObject>> {
+  return manager.ready.then(() => ({
```

Comment:
We don't want the app to wait for the service manager to be ready, it could be a while if there are a lot of kernel specs on a slow connection.